### PR TITLE
support binary schemas as flatc input

### PR DESCRIPTION
**Pull request by fbenkstein**
Support binary schemas in flatc and flatbuffers::Parser.  This can
simplify loading JSON data in some cases since there's no issue with
include files.

### CODE REVIEW

File: src/idl_parser.cpp [hunk 1]
```diff
@@ -2530,6 +2530,12 @@ std::set<std::string> Parser::GetIncludedFilesRecursive(
 
 // Schema serialization functionality:
 
+static std::string UnqualifiedName(std::string name) {
```
Comment (VeaaC) [inline src/idl_parser.cpp]: This could fail badly in case the filename does not have a dot, but the foldername has?

File: src/idl_parser.cpp [hunk 2]
```diff
@@ -2579,6 +2585,112 @@ void Parser::Serialize() {
   }
 }
 
+static Namespace *GetNamespace(
+    const std::string &qualified_name, std::vector<Namespace *> &namespaces,
+    std::map<std::string, Namespace *> &namespaces_index) {
+  const char *dot = strrchr(qualified_name.c_str(), '.');
+  std::string namespace_name =
+      dot ? namespace_name.assign(qualified_name.c_str(), dot) : "";
```
Comment (VeaaC) [inline src/idl_parser.cpp]: using namespace_name to construct itself?

File: src/idl_parser.cpp [hunk 3]
```diff
@@ -2579,6 +2585,112 @@ void Parser::Serialize() {
   }
 }
 
+static Namespace *GetNamespace(
+    const std::string &qualified_name, std::vector<Namespace *> &namespaces,
+    std::map<std::string, Namespace *> &namespaces_index) {
```
Comment (VeaaC) [inline src/idl_parser.cpp:2676]: unique_ptr for tracking Namespace ownership??

File: src/idl_parser.cpp [hunk 4]
```diff
@@ -2579,6 +2585,112 @@ void Parser::Serialize() {
...
+bool Parser::Deserialize(const reflection::Schema *schema) {
+  file_identifier_ = schema->file_ident() ? schema->file_ident()->str() : "";
+  file_extension_ = schema->file_ext() ? schema->file_ext()->str() : "";
+  std::map<std::string, Namespace *> namespaces_index;
+
+  // Create defs without deserializing so references from fields to structs and
+  // enums can be resolved.
+  for (auto it = schema->objects()->begin(); it != schema->objects()->end();
+       ++it) {
+    auto struct_def = new StructDef();
```
Comment (VeaaC) [inline src/idl_parser.cpp:2709]: Does Flatbuffer not use unique_ptr?

File: src/idl_parser.cpp (hunk 2 above)
Comment (fbenkstein) [reply src/idl_parser.cpp]: No the qualified name is the qualified type name.  The namespace name is everything before the last dot.

File: src/idl_parser.cpp (hunk 4 above)
Comment (fbenkstein) [reply src/idl_parser.cpp:2709]: It doesn't, unfortunately.

**Pull request closed without merging**